whenchanged/whenchanged.py

```diff
@@ -135,49 +137,62 @@ def run(self):
 def print_usage(prog):
     print(__doc__ % {'prog': prog}, end='')
 
+def parse_args(argv):
+    """
+    Parse the various args and returns a dictionnary.
 
-def main():
-    args = sys.argv
-    prog = os.path.basename(args.pop(0))
-
-    if '-h' in args or '--help' in args:
-        print_usage(prog)
-        exit(0)
+    The function returns None if the arguments were asking for the help.
 
+    The function raises ValueError in case of bad arguments.
+    """
     files = []
     command = []
     recursive = False
     verbose = False
     run_once = False
     run_at_start = False
 
-    while args and args[0][0] == '-':
-        flag = args.pop(0)
-        if flag == '-v':
+    prog = os.path.basename(argv.pop(0))
+
+    # pre-parsing
+    # -c is a very special case: all following arguments are part of a command and should not be parsed
+    argv_until_c = argv
+    for i, a in enumerate(argv):
+        m = re.match(r'\-[^\-c]*c(.?)', a)
+        if m:
+            split_i = i + 1
+            if not m.group(1):
+                # argument not attached to the option, give it to the parser
+                split_i += 1
+            argv_until_c, command = argv[:split_i], argv[split_i:]
+            break
+
+    args, shiftargs = gnu_getopt(argv_until_c, 'hvr1sc:', longopts=["help"])
+    for arg,argopt in args:
+        if arg == '-h' or arg == '--help':
+            print_usage(prog)
```

Comment:
I wonder if it's better to call print_usage() outside this function, i.e. in main. Would be nice to test `parse_args` without it printing stuff to stdout.